web/src/components/navbar/NavBarPopupMenu.test.tsx

```diff
@@ -0,0 +1,281 @@
+import { NavBarPopupMenu, Props as NavBarPopupMenuProps } from "@/components/navbar/NavBarPopupMenu";
+import { getMergedConfig } from "@/contexts/configContext";
+import * as api from "@/lib/api-client/apiClient";
+import { IsAuthenticated } from "@/lib/auth/AuthContext";
+import { triggerSignIn } from "@/lib/auth/sessionHelper";
+import { onSignOut } from "@/lib/auth/signinHelper";
+import { ROUTES } from "@/lib/domain-logic/routes";
+import { withAuth } from "@/test/helpers/helpers-renderers";
+import { mockPush, useMockRouter } from "@/test/mock/mockRouter";
+import { useMockUserData } from "@/test/mock/mockUseUserData";
+import { setupStatefulUserDataContext, WithStatefulUserData } from "@/test/mock/withStatefulUserData";
+import {
+  BusinessUser,
+  generateProfileData,
+  generateUser,
+  generateUserData,
+  UserData,
+} from "@businessnjgovnavigator/shared";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+
+const Config = getMergedConfig();
+
+const mockApi = api as jest.Mocked<typeof api>;
+
+jest.mock("next/router", () => ({ useRouter: jest.fn() }));
+jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
+jest.mock("@/lib/data-hooks/useRoadmap", () => ({ useRoadmap: jest.fn() }));
+jest.mock("@/lib/api-client/apiClient", () => ({ postSelfReg: jest.fn() }));
+jest.mock("@/lib/auth/signinHelper", () => {
+  const originalSigninHelper = jest.requireActual("@/lib/auth/signinHelper");
+  return {
+    ...originalSigninHelper,
+    onSignOut: jest.fn(),
+  };
+});
+jest.mock("@/lib/auth/sessionHelper", () => ({
+  triggerSignIn: jest.fn(),
+}));
```

Comment:
So I mocked out, triggerSignIn and onSignOut, because they're tested in other parts of the code base independently. 

I'd already programmed the register section at that point so I figured I'd leave it but I can mock that out as well if this is bad for inconsistency reasons. 

I'm assuming mocks are fine because they're tested in depth in other places and these are unit tests but if others think I'm wrong I would love to know